### PR TITLE
Fridges will now have their door blow off if hit by ex_act(1)

### PR DESCRIPTION
**Pull request by ghost**
- If a fridge is hit by ex_act(1), its door will now be blown off. This makes the fridge permanently useless
- But hey, at least you're fine, and so is everything in the fridge except the fridge itself (closets normally give you ex_act 2 if hit by ex_act 1, which hurts a whole lot)
- You can't fix the bridge, but you can destroy it with a welding tool

Fully tested

### CODE REVIEW

**Comment by Hoshi-chan** [comment]
Stop removing anything that interferes with your vision of fucking meteor rounds, holy shit.

**Comment by TheFriggenSawbones** [comment]
i will comment out meteors don't you dare

**Comment by Probe1** [comment]
On one hand its a dylan-ism PR.  On the other hand a reference to objectively the worst Indiana Jones movie and arguably the biggest disappointment of that year is dumb.

**Comment by AutismSpider** [comment]
>oh no I can't power-game meteor rounds and hide in a fridge

man the fuck up and set up the shield gens OH but wait no one knows how to do this any way

**Comment by Kurfursten** [comment]
@Probe1 Red Letter Media liked that one scene so that means the internet has to like it, by /tv/ rules

**Comment by ghost** [comment]
What if, instead, explosions threw the fridge away from the blast, akin to when you're in space and an explosion goes off nearby, throwing you faster than an admin can try to explode you.

You'll survive the meteors, but you'll go flying rather than be some static object.

**Comment by Kurfursten** [comment]
@Skullyton That's pretty fun. I was in a round where Sawbones immediately ran to a freezer so the admins threw it into space

**Comment by ghost** [comment]
@Kurfursten That might've been me tossing it around. 

It was our third meteor round that day IIRC as dylan was testing it/the admins wanted to torture the playerbase, so to spice it up, threw in some irregularities such as tossing the fridge around that everyone was in, save it from being 'sit in the fridge in a big old group for 30 minutes'

Some art spawned from that IIRC

**Comment by ghost** [comment]
> adminbussed a meteor round without a vote
> After it was voted down last round
> Gets out fukken fridged
> Gets so mad [19:07:23]ADMIN: Dylanstrategie/(Henry Alouette) deleted Meat Fridge at (262,255,1) when he wasted a half hour of my life on memetors

Nah, I'm fine with actually being able to test more than one canister bomb a round

**Comment by ghost** [comment]
If you seriously think I went to do that because of the latest Meteors round, you're straight up delusional

I had already noticed fridges were used for powergaming due to a meme reference that wasn't thought up months ago, along with laying in space which I also tried to fix but apparently that didn't work so I need to look at it

It goes to the point where I pre-emptively deleted the Kitchen's fridge, then I deleted another one I noticed later and what do you know there's people sitting in it

This PR still makes it so you perfectly survive an explosion if sitting in a fridge. ONE explosion

**Comment by ghost** [comment]
Pure salt coding. No thanks. I'd like to be able to knick the fridge and test canister bomb mixes without waiting 40 minutes for the borgs and medbay I pda'd before the test to pick my bombsuited body out cause I can't fridge test a bunch of mixes anymore.

AutismSpider, it's not that hard but since the meteors are in tightly packed waves now even dual offset shield gens get punched. Still need to grille and wall up a couple layers behind it until you can get more circuits from cargo and a miner to actually go mine and get minerals to the rnd in eng foyer that chuted under the shield. It gets easier once you have a third offset shield on eng and 2 on the mining dock to start pushing back.

**Comment by ghost** [comment]
The fridge will still keep you perfectly safe for one direct explosion

Also regarding Meteors balance this is another thing I want to touch on much later, notably because shield generators are terrible right now. However, I do not want to sneak in fridge changes to my Meteors PR because that's bad, said the same people who clearly never want their comfort zone switched

**Comment by ghost** [comment]
Ever done toxins? Getting research done in round two "valves" at a time with the second killing you is pretty shit. I'd have to look and maybe one of the others is a child but that'd still only be three.

**Comment by ghost** [comment]
If you think that canister bombs are that hard to test, it's another issue

**Comment by ghost** [comment]
How so, cause you gotta be adjacent so you're in that blast one way or another. You can keep your head on with a bombsuit/AEOD but only the fridge or getting some schmuck to open it for you is gonna test it.

Just up the maximum amount the shield capacitor can charge to 3 times what it is and see how that goes. That's been the limiter on the shield's size and strength. So we still need to power it but it's more practical.

**Comment by ghost** [comment]
If this is gonna happen, why not make fridges buildable?

It'd make keeping organs that decay such as brains alive easier

**Comment by ghost** [comment]
You can already order fridges if you're ready to have meat in them

The problem with buildable fridges is that it would put us back to square one, as soon as the fridge is blown up a dozen more would be made out of metal or plasteel

**Comment by ghost** [comment]
You got the freezer crate in med, the mmi's in robo, and you don't need to save the monkey brain in a ghetto surgery swap. What's the situation they're keeling on you?

**Comment by 9600bauds** [comment]
But I thought frenchmen loved cheese

**Comment by Allspice** [comment]
Remove fridges remove meteors everyone wins.

**Comment by TheFriggenSawbones** [comment]
I like a good Gouda personally

**Comment by xpcybic** [comment]
This does seem like a necessary change to keep meteors from being "hide in the fridge for 20 minutes: the gamemode" but you need to also make meteors more reasonable to deal with or exactly zero people will appreciate this. Breaks in the meteor waves, fewer meteors, _something_.

**Comment by Probe1** [comment]
To be fair to Dylan there are two issues here being conflated into one.  The fridge thing was discussed before the meteor thing happened.

**Comment by ghost** [comment]
The problem with Meteors is that I don't have a ton of room to tweak values and what not, right now there's only a few sliders I can use : 
- The amount of meteors per wave and the spread (Pretty big right now)
- The delay between waves (Should be 30-40 seconds currently)
- The length of the "build time" (Currently 7 and a half to 10 minutes)
- The length until the shuttle arrives (Currently 20 to 22 and a half minutes)
- The damage done by each meteor type

My plan is to build around that by granting Engineering some brand new and updated toys to counter meteor waves, that they either need to set up or that may even be built into the map, and by updating the wave logic by adding more special waves, both to counter camping and give Engineering longer breaks

At the end of the day, I need to balance all of this so the station ends up pulverized if nothing is done, and you can save about everyone if serious work is done. Harder than it sounds, Engineers can only do so much in ten minutes, especially with the shielding that is so weak

**Comment by Shadowmech88** [comment]
We could just remove the fridge entirely and use all this _salt_ to preserve the meat.

**Comment by AutismSpider** [comment]
@SarahJohnson489  can order more shields form cargo no one never does any way

**Comment by ghost** [comment]
Did we ever get fulls? I just remembered it being circuits that needed advanced stock parts

edit: xpcybic, a quick fix would be either allowing a shield generator to accept multiple capacitors or raising the charge limit on the capacitors so the shield generators are a more practical choice. When it's more user friendly to hunker down in eng or an ame cargonia everything can start to come together for the mode. People will become more on board with the meta of quickly moving everything over to the bunker and setting up chutes to move through shields. 

Another I suppose would be tweaking the max distance the telesci/teleporter shield gens can stretch their shield so you'd have a long wall that you'd have to harden a range of around 8 tiles to save the wire powering it.

A lot of the problem with the mode is few being on the same page of round meta and general lack of experience with construction repairs, and the machinery involved which can be partly cured with exposure.

**Comment by NigglyWiggly** [comment]
Jesus fucking Christ, man.

The problem with meteors is that it's just not fun.
It will be very hard to MAKE it fun.
Removing this and the ability to rest to matrix-dodge meteors isn't going to make it suddenly more fun because now you can't avoid dying in two specific ways.

**Comment by ghost** [comment]
but why

**Comment by Shadowmech88** [comment]
The main problem with meteors, I think, is that there is very little that non-engineers can do against meteors.
So for the engineers it becomes "work nonstop to fortify with shield generators/patch damage" and for everyone else it becomes "hide in the bar while the engineers work nonstop to fortify with shield generators/patch damage."

**Comment by ghost** [comment]
He can't get his meteor murderboner wet when people hide in the fridge is why celt.

Everyone else not mining should be moving into eng and those with suits building and dying to maintain the perimeter walls behind the shields. Others not stepping up to move things and to reinforce really hurts it.

**Comment by 9600bauds** [comment]
"Controversial"

**Comment by clusterfack** [comment]
Merge it baby

**Comment by NigglyWiggly** [comment]
no

File: code/game/objects/structures/crates_lockers/closets/secure/freezer.dm [hunk 1]
```diff
@@ -13,18 +16,43 @@
...
 /obj/structure/closet/secure_closet/freezer/ex_act(var/severity)
-	// IF INDIANA JONES CAN DO IT SO CAN YOU
 
-	// Bomb in here? (using same search as space transits searching for nuke disk)
+	//Bomb in here? (using same search as space transits searching for nuke disk)
 	var/list/bombs = search_contents_for(/obj/item/device/transfer_valve)
 	if(!isemptylist(bombs)) // You're fucked.
 		..(severity)
 
+	if(severity == 1)
```
Comment (ComicIronic) [inline code/game/objects/structures/crates_lockers/closets/secure/freezer.dm:34]: I can't believe it's not defines.
This won't block your PR, but it's still nonsense.

Comment (ghost) [inline code/game/objects/structures/crates_lockers/closets/secure/freezer.dm:34]: Well, there aren't any defines for ex_act(). It's just oldcode to you

All the switch instructions I've seen just use magic numbers. 1, 2, 3

**Comment by Imagebard** [comment]
Eh, I don't see the problem with this PR. Sarah is the only one with a good reason not to merge it, everyone else is,well...

>Stop removing anything that interferes with your vision of fucking meteor rounds, holy shit.

What kind of mode in existence should consist of you hiding in a locker until the shuttle comes? Is that your idea of fun or something?

**Comment by Shadowmech88** [comment]
> How the hell can we test the gamemode if people just grab a space suit and lie down/in a locker?

The gamemode has been tested a decent number of times already, unless you're suggesting that _literally_ every player in every round of Meteor always hides in one of the three fridges on the station and/or manages to get a spacesuit, and so all the previous rounds shouldn't count as tests.

I think the core concept of Meteor, which is "stay alive for 15 minutes as meteors hit the station" is flawed.

It may be possible to make a new gamemode in which meteor showers play a more prominent role than usual, but I don't feel like meteors being the primary antagonist of a gamemode is something that most players will ever find fun.

**Comment by Imagebard** [comment]
>I think the core concept of Meteor, which is "stay alive for 15 minutes as meteors hit the station" is flawed.

But this PR isn't changing the core concept of Meteor, its about a exploit in it. Its like you were able to completely stop nuclear operatives form getting the disk by putting it in a invulnerable locker. Such exploits are stupid because you can just ghost instead of

If you want to stop the Meteors game mode in first place, you should go complain about meteors, not a bugfix for a horrendous exploit that easily makes the mode casual tier.

**Comment by Shadowmech88** [comment]
Last time I checked, Meteor isn't even in rotation, it has to be _bussed_ in.
I don't approve of nerfing a feature just because that feature can be used to exploit a single very uncommon game mode, that isn't even in rotation, and that very few people even like.

**Comment by Imagebard** [comment]
As much as the rarity is true, I don't see why jumping into fridges should make you immune to explosions in first place.

**Comment by AutismSpider** [comment]
Locker use to make you immune to explosion and damage that fix fast. 

Same should be done for fridges

**Comment by clusterfack** [comment]
Not really sure what to do about this since all the reactions are memevotes and all the complaints are about meteors and not about this PR in any way.

**Comment by ghost** [comment]
The only use of fridges being immune to explosions that has some gameplay legitimacy I've seen is canister bombs

This could be solved by implementing basic remote signalers to canisters (think like airlocks where you can install signalers and then remotely bolt or electrify them)

**Comment by 9600bauds** [comment]
Great job, remote signalers now work like pseudo-TTVs.

**Comment by ghost** [comment]
Don't see a problem with that

**Comment by ghost** [comment]
Absolutely do not want a way to remote det canister bombs. You can make ten bajillion of them and it only takes roughly 10 minutes to get the basic mix for bombcap in any of a number of locations from which you can load and send them off as fast as you can get metal for cans, tank dispensers for tanks, and jihadists to use them. There's also the issue with all the maint hideyholes and maint traps in general. You can get around cap by just putting 15 of them all over the station in rarely checked spots on top of a prox-signaller so if anyone tries to disarm the station gets poof'd. Essentially bombcap versions of the ol' prox fuel tank in maint. 

Arguably a balance issue against malf/blob and shits on wizlingpire. They're mega fuel tanks and if the AI/blob is busy elsewhere that it misses you at the core, telesci or that you snuck a beacon nearby you can repeatedly bomb it with cans. Mind that 28x52 or roughly 3-4 escapes worth of non-blob core or shield wall getting repeatedly blown out leaves little to defend against 1 madman with two bags of guns and a jetpack to push them over. You can tear though a lot of ground quick as well as make large gaps that are costly for blob to cross.  It'd no longer be oh hey you can only drop 6 bombs. No, you can bomb every single inch of it without losing a single life once you pull your telesci sheet up or have enough guys to run cans and keep up pressure.

You can hit bombcap with extended emergency oxygen tanks and a wide range of mixes too so it's not even like oh some thefts and breakins will be noticed. 

I suppose what I'm saying is it sidesteps the chance where people would be caught and it takes the breaks off what antags can safely do with explosives. Like stealing the valves or rigging 6 with 0 tests. Because of all the places they can be made, it's not easy to watch out for. Essentially every use where you'd say fuel tank can be replaced with 10 minutes and bombcap. Every time you'd say you would have suicide bombed to cut a quadrant of the blobbed part of the station off it'd be replaced with just bombed then kept on going. And it makes preventing bombcap explosions pretty difficult after a couple infodumps if you're malf/blob/wiz/pire that were trying to get the valves out of the round so your big threat is gone.

While someone could make an "explosive tank fracture inducer" a research item to work as a signaller for it I'm still not a fan because it's still unlimited "ttv's" without risk to the user or it needs exotics and you have to go mine yourself. A plasteel "bomb bunker" or researchable bomb locker wont work cause it needs exotics, research that requires exotics or using telesci on telcomms, too many minerals or people can just rush it on meteors and it's back to square one. I guess if you made a ttv with a device in it attachable to a canister with an overlay and removable with a wrench and says so in the canister descrip? It's not the best solution, pretty illogical but it works as a compromise I guess. Maybe intermediate steps to "modify" the ttv so it's less of a leap of logic.

I'd say side step the issue and making meteor more interactive for people who aren't engineers. Though how is the golden question.

You could increase the power storage cap/charge rate on the shield capacitors. If they can actually hold a location better then a meta can build around getting that setup and everyone into it to reinforce so they can fight to survive instead of playing dodge the space ruk that explodes through plasteel rapidly. Once that's solidly established and people are commonly thriving then the issue of the round ends in x minutes it's all pointless anyway can be tackled. The people who didn't want to play will still hop in the fridge for 45 minutes so they can do what they came to play before finding out it was meteors. They didn't want to do meteors anyway and probably voted no or missed the vote.

tfw 6 bombcap bombs and 35 bombcap canister bombs

**Comment by ghost** [comment]
I already have plans for Meteors, but I'm currently forced to put any ambitious plans on stand-by waiting for Wenesday

I'm perfectly aware the three main problems with Meteors are player and ghost interactivity, means of defending the station efficiently that need effort but with great reward and getting rid of cheeses like hiding in the Core, laying in space (still works for some reason, need to give it a more thorough look) and fridges

At the end of the day, the fridge issue has been brought up due to Meteor testing, but it's a general thing. You could conceive someone using a Fridge to casually throw bombs at very close range and survive them

**Comment by Shadowmech88** [comment]
> You could conceive someone using a Fridge to casually throw bombs at very close range and survive them

If this was a big problem, I think it would've been complained about at some point during the _years_ since fridges were made explosion-proof.

**Comment by AutismSpider** [comment]
@Shadowmech88 this happen with locker in the past they were explosion proof what happen to lockers should have happen to the fridge to

**Comment by Imagebard** [comment]
@Shadowmech88 
As was noted by AutismSpider, lockers were once explosion proof, and that was nerfed. So what happens here is that Fridge being immune to explosions might be a oversight.

**Comment by ihadtoregisterforthis** [comment]
Fridge bring immune to explosions sounds like a feature

**Comment by Imagebard** [comment]
Because of the reference you mean?

**Comment by AutismSpider** [comment]
Reference to bad movie

**Comment by 9600bauds** [comment]
It's specifically coded in, it's not an oversight

**Comment by ghost** [comment]
And it has not been completely removed. You can survive a dev in the Fridge, but it will be broken. Other explosions won't even scratch the fridge

**Comment by AutismSpider** [comment]
so its one use thing.

Whats the big deal then?

**Comment by ghost** [comment]
It was added back in specifically imagebard. I forget why Nex put it in but it was a great day for being able to test a bunch of canister bomb mixes a round so you can get irritated at how much you load and the starting temp skewing what's an "ideal" ratio into just methodically test all the combos you can before shuttle.

**Comment by Imagebard** [comment]
Just stand out of the dev1 blast radius. Is being within the center of the bomb really necessary? 
Actually, don't we have the bhangometer to see the explosion radius?

Can you explain to me why being so close to the bomb is necessary?

**Comment by 9600bauds** [comment]
How are you going to remotely open the release valve on a canister

**Comment by Imagebard** [comment]
Oh, Canister bombs
Well fuck I actually have no answer.

**Comment by ComicIronic** [comment]
@clusterfack gets final say

**Comment by clusterfack** [comment]
Also *almost* none of this shit is about the change and all of it is about meteors so I don't really think your  complaints about an unrelated topic are valid

**Comment by unid15** [comment]
http://niceme.me

**Comment by kol1th** [comment]
> merged

:eggplant: :eggplant: :100: :ok_hand:

**Comment by ghost** [comment]
It gets wrecked by being thrown about, and that fridge door is still on by the time it lands.

**Comment by clusterfack** [comment]
Suck my merge, choke on it

#philthecollab

**Comment by ghost** [comment]
Still say the explosion should throw the fridge so you're thrown off Z level if it's meteors

Fridge-based projectile space exploration when?

**Comment by NigglyWiggly** [comment]
@Skullyton No we can't have that, that'd be fun.

God knows we can't have fun here, especially where meteors are concerned

**Comment by TheFriggenSawbones** [comment]
working on my PR right now

**Comment by Shadowmech88** [comment]
> 2 likes
> 19 dislikes
> merged anyway

Simbly ebin.

**Comment by 9600bauds** [comment]
Clusterfack has a votes problem, please don't make fun of him

**Pull request merged**